src/components/ArticlePage.jsx

```diff
@@ -1,28 +1,46 @@
 import { useState } from "react";
-import { fetchArticlesById, fetchArticleComments } from "./api";
+import { fetchArticlesById, fetchArticleComments, patchArticle } from "./api";
 import { useEffect } from "react";
 import { useParams } from "react-router-dom";
 import { CommentCard } from "./CommentCard";
 
 export function ArticlePage() {
   const [article, setArticle] = useState();
   const [comments, setComments] = useState();
+  const [votes, setVotes] = useState();
+
   const { article_id } = useParams();
 
   useEffect(() => {
     fetchArticlesById(article_id).then((article) => {
       setArticle(article);
+      setVotes(article.votes);
     });
     fetchArticleComments(article_id).then((comments) => {
       setComments(comments);
     });
   }, [article_id]);
+
+  function handleClick(event) {
+    const inc = event.target.value;
+    patchArticle(article_id, inc).then((data) => {
+      setVotes(data.article.votes);
+    });
+  }
```

Comment:
Nice, this works really well, but this can be done optimistically, which I would recommend as it is a good example of a case when optimistic rendering is done.